browser/ui/BUILD.gn

```diff
@@ -311,6 +311,8 @@ source_set("ui") {
       "//brave/browser/profiles:util",
       "//brave/browser/ui/sidebar",
       "//brave/browser/ui/tabs:split_view",
+      "//brave/browser/ui/tabs:tab_model",
```

Comment:
why is this split into tab_model and tab_model_impl if they are both included here and nowhere else?

Reply:
Right, it's not necessary now. updated to use `tab_model` only. we could do it later when it needs.